# Bartelo-ANGSD-Final-Project

What each resource below tells us:

1. Overall use of animals in diabetic research and why db/db is the best.

2. Diabetes in humans

3. Paper we are basing the whole project off of

4. Different paper not necessary to include

5. Different paper not necessary to include

6. Where we get the samples from 

7. Explains why we used some of the STAR parameters we did - e.g. sjdbOverhang = # of bp - 1

### Resources

1. King, Aileen J F. “The Use of Animal Models in Diabetes Research.” British Journal of Pharmacology, Blackwell Publishing Ltd, June 2012, www.ncbi.nlm.nih.gov/pmc/articles/PMC3417415/#:~:text=Although%20rats%20and%20mice%20are,to%20develop%20obesity%20in%20captivity. 

2. Fuchsberger, Christian, et al. “The Genetic Architecture of Type 2 Diabetes.” Nature, U.S. National Library of Medicine, 2017, www.ncbi.nlm.nih.gov/pmc/articles/PMC5034897/#SD1. 

3. Matthews, DR., et al. “RNA-Seq Analysis of Islets to Characterise the Dedifferentiation in Type 2 Diabetes Model Mice Db/Db.” Endocrine Pathology, Springer US, 1 Jan. 1998, link.springer.com/article/10.1007/s12022-018-9523-x. 

4. Goyal, Neha, et al. “RNA Sequencing of Db/Db Mice Liver Identifies LncRNA H19 as a Key Regulator of Gluconeogenesis and Hepatic Glucose Output.” Scientific Reports, Nature Publishing Group UK, 16 Aug. 2017, www.ncbi.nlm.nih.gov/pmc/articles/PMC5559625/. 

5. Zhang, Fang, et al. “Gene Expression Profile Analysis of Type 2 Diabetic Mouse Liver.” PLOS ONE, Public Library of Science, 2013, journals.plos.org/plosone/article?id=10.1371%2Fjournal.pone.0057766. 

6. “GEO Accession Viewer.” National Center for Biotechnology Information, U.S. National Library of Medicine, www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE107489. 

7. Dobin, Alexander, and Thomas Gingeras. “Optimizing RNA-Seq Mapping with STAR.” Methods in Molecular Biology (Clifton, N.J.), U.S. National Library of Medicine, 2016, pubmed.ncbi.nlm.nih.gov/27115637/. 

